src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -38,23 +46,181 @@
 @RequiredArgsConstructor
 @Log4j2
 public class MLCommonsClientAccessor {
-    private static final List<String> TARGET_RESPONSE_FILTERS = List.of("sentence_embedding");
+
+    /**
+     * Inference parameters for calls to the MLCommons client.
+     */
+    public static class InferenceRequest {
+
+        private static final List<String> DEFAULT_TARGET_RESPONSE_FILTERS = List.of("sentence_embedding");
+
+        private final String modelId;
+        private final List<String> inputTexts;
+        private final MLAlgoParams mlAlgoParams;
+        private final List<String> targetResponseFilters;
+        private final Map<String, String> inputObjects;
+        private final String queryText;
+
+        public InferenceRequest(
+            @NonNull String modelId,
+            List<String> inputTexts,
+            MLAlgoParams mlAlgoParams,
+            List<String> targetResponseFilters,
+            Map<String, String> inputObjects,
+            String queryText
+        ) {
+            this.modelId = modelId;
+            this.inputTexts = inputTexts;
+            this.mlAlgoParams = mlAlgoParams;
+            this.targetResponseFilters = targetResponseFilters == null ? DEFAULT_TARGET_RESPONSE_FILTERS : targetResponseFilters;
+            this.inputObjects = inputObjects;
+            this.queryText = queryText;
+        }
+
+        public String getModelId() {
+            return modelId;
+        }
+
+        public List<String> getInputTexts() {
+            return inputTexts;
+        }
+
+        public MLAlgoParams getMlAlgoParams() {
+            return mlAlgoParams;
+        }
+
+        public List<String> getTargetResponseFilters() {
+            return targetResponseFilters;
+        }
+
+        public Map<String, String> getInputObjects() {
+            return inputObjects;
+        }
+
+        public String getQueryText() {
+            return queryText;
+        }
+
+        /**
+         * Builder for {@link InferenceRequest}. Supports fluent construction of the request object.
+         */
+        public static class Builder {
```

Comment:
Lots of this builder code can be collapsed if we use Builder annotation from Lombok. Something like this:

```
@Getter
@Builder
public class InferenceRequest {

    private final String modelId; // required
    @Singular("inputText")
    private List<String> inputTexts;
    private MLAlgoParams mlAlgoParams;
    private List<String> targetResponseFilters;
    private Map<String, String> inputObjects;
    private String queryText;

    private InferenceRequest(String modelId, List<String> inputTexts, MLAlgoParams mlAlgoParams, 
                             List<String> targetResponseFilters, Map<String, String> inputObjects, String queryText) {
        this.modelId = modelId;
        this.inputTexts = inputTexts != null ? inputTexts : new ArrayList<>();
        this.mlAlgoParams = mlAlgoParams;
        this.targetResponseFilters = targetResponseFilters;
        this.inputObjects = inputObjects;
        this.queryText = queryText;
    }

    public void addInputText(String inputText) {
        if (this.inputTexts == null) {
            this.inputTexts = new ArrayList<>();
        }
        this.inputTexts.add(inputText);
    }
}
```

and call it in the similar way:
```
InferenceRequest request2 = InferenceRequest.builder()
        .modelId("model123")
        .inputText("Sample text 1")
        .inputText("Sample text 2")
        .inputText("Sample text 3")
        .build();
```